scripts/install_deps.ps1

```diff
@@ -6,23 +6,26 @@ $progressPreference = "silentlyContinue"
 if ( -not (Test-Path "$PSScriptRoot\..\deps\boost") ) {
   New-Item -ItemType Directory -Force -Path "$PSScriptRoot\..\deps"
 
-  Invoke-WebRequest -URI "https://github.com/Kitware/CMake/releases/download/v3.18.2/cmake-3.18.2-win64-x64.zip" -OutFile cmake.zip
-  if ((Get-FileHash cmake.zip).Hash -ne "5f4ec834fbd9b62fbf73bc48ed459fa2ea6a86c403106c90fedc2ac76d51612d") {
+  Invoke-WebRequest -URI "https://github.com/Kitware/CMake/releases/download/v3.27.4/cmake-3.27.4-windows-x86_64.zip" -OutFile cmake.zip
+  if ((Get-FileHash cmake.zip).Hash -ne "e5e060756444d0b2070328a8821c1ceb62bd6d267aae61bfff06f96c7ec943a6") {
     throw 'Downloaded CMake source package has wrong checksum.'
   }
   tar -xf cmake.zip
-  mv cmake-3.18.2-win64-x64 "$PSScriptRoot\..\deps\cmake"
+  mv cmake-3.27.4-windows-x86_64 "$PSScriptRoot\..\deps\cmake"
+  Remove-Item cmake.zip
 
   # FIXME: The default user agent results in Artifactory treating Invoke-WebRequest as a browser
   # and serving it a page that requires JavaScript.
-  Invoke-WebRequest -URI "https://boostorg.jfrog.io/artifactory/main/release/1.77.0/source/boost_1_77_0.zip" -OutFile boost.zip -UserAgent ""
-  if ((Get-FileHash boost.zip).Hash -ne "d2886ceff60c35fc6dc9120e8faa960c1e9535f2d7ce447469eae9836110ea77") {
+  Invoke-WebRequest -URI "https://boostorg.jfrog.io/artifactory/main/release/1.83.0/source/boost_1_83_0.zip" -OutFile boost.zip -UserAgent ""
+  if ((Get-FileHash boost.zip).Hash -ne "c86bd9d9eef795b4b0d3802279419fde5221922805b073b9bd822edecb1ca28e") {
     throw 'Downloaded Boost source package has wrong checksum.'
   }
   tar -xf boost.zip
-  cd boost_1_77_0
+  Remove-Item boost.zip
```

Comment:
Isn't the windows machines using bash? Interesting enough it seems that bash and power-shell commands works.

Reply:
They're using whatever shell we choose. If you run a PowerShell script, it runs under PowerShell.

And there are no Bash commands here. `cd` is apparently an alias for `Set-Location` and `tar` is just `tar.exe` - probably installed along with [Git Bash](https://gitforwindows.org/).

But yeah, given that CircleCI hash Bash installed, would be nice to have this written in Bash. Looking up PowerShell syntax is always a pain whenever I have to modify it. The main problem with it would be that this script is [documented as a way to install deps on Windows](https://docs.soliditylang.org/en/latest/installing-solidity.html#prerequisites-windows) and I'm not sure we can assume everyone has Bash installed.

Reply:
> They're using whatever shell we choose. If you run a PowerShell script, it runs under PowerShell.
> 
> And there are no Bash commands here. `cd` is apparently an alias for `Set-Location` and `tar` is just `tar.exe` - probably installed along with [Git Bash](https://gitforwindows.org/).
> 
> But yeah, given that CircleCI hash Bash installed, would be nice to have this written in Bash. Looking up PowerShell syntax is always a pain whenever I have to modify it. The main problem with it would be that this script is [documented as a way to install deps on Windows](https://docs.soliditylang.org/en/latest/installing-solidity.html#prerequisites-windows) and I'm not sure we can assume everyone has Bash installed.

Oh, right. I assumed that it was using bash as default shell because of this: https://github.com/ethereum/solidity/blob/develop/.circleci/config.yml#L561

Because I needed to add it to make the matrix notification to work on any OS. So we can probably migrate this script to bash as well. But this is out-of-scope anyway.